third_party/terraform/resources/resource_storage_transfer_job.go

```diff
@@ -0,0 +1,529 @@
+	return nil
+}
+
+func resourceStorageTransferJobRead(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+
+	project, err := getProject(d, config)
+	if err != nil {
+		return err
+	}
+
+	name := d.Get("name").(string)
+	res, err := config.clientStorageTransfer.TransferJobs.Get(name).ProjectId(project).Do()
+	if err != nil {
+		return handleNotFoundError(err, d, fmt.Sprintf("Transfer Job %q", name))
+	}
+	log.Printf("[DEBUG] Read transfer job: %v in project: %v \n\n", res.Name, res.ProjectId)
+
+	d.Set("project", res.ProjectId)
+	d.Set("description", res.Description)
+	d.Set("status", res.Status)
+	d.Set("last_modification_time", res.LastModificationTime)
+	d.Set("creation_time", res.CreationTime)
+	d.Set("deletion_time", res.DeletionTime)
+
+	err = d.Set("schedule", flattenTransferSchedules([]*storagetransfer.Schedule{res.Schedule}))
+	if err != nil {
+		return err
+	}
+
+	d.Set("transfer_spec", flattenTransferSpecs([]*storagetransfer.TransferSpec{res.TransferSpec}))
+	if err != nil {
+		return err
+	}
+
+	jobId, err := extractTransferJobId(res.Name)
+	if err != nil {
+		fmt.Printf("Error extracting transfer job id %v: %v", name, err)
+		return err
+	}
+
+	log.Printf("[DEBUG] Patched transfer job: %v\n\n", jobId)
+	d.SetId(jobId)
+	return nil
+}
+
+func resourceStorageTransferJobDelete(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+
+	project, err := getProject(d, config)
+	if err != nil {
+		return err
+	}
+
+	transferJobName := d.Get("name").(string)
+
+	transferJob := &storagetransfer.TransferJob{
+		Status: "DELETED",
+	}
+
+	fieldMask := "status"
+
+	updateRequest := &storagetransfer.UpdateTransferJobRequest{
+		ProjectId:   project,
+		TransferJob: transferJob,
+	}
+
+	updateRequest.UpdateTransferJobFieldMask = fieldMask
+
+	// Update transfer job with status set to DELETE
+	log.Printf("[DEBUG] Setting status to DELETE for: %v\n\n", transferJobName)
+	err = resource.Retry(1*time.Minute, func() *resource.RetryError {
+		_, err := config.clientStorageTransfer.TransferJobs.Patch(transferJobName, updateRequest).Do()
+		if err != nil {
+			return resource.RetryableError(err)
+		}
+		if gerr, ok := err.(*googleapi.Error); ok && gerr.Code == 429 {
+			return resource.RetryableError(gerr)
+		}
+		return nil
+	})
+
+	if err != nil {
+		fmt.Printf("Error deleting transfer job %v: %v\n\n", transferJob, err)
+		return err
+	}
+
+	log.Printf("[DEBUG] Deleted transfer job %v\n\n", transferJob)
+
+	return nil
+}
+
+func resourceStorageTransferJobStateImporter(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+	parts := strings.Split(d.Id(), "/")
+	switch len(parts) {
+	case 2:
+		d.Set("project", parts[0])
+		d.Set("name", fmt.Sprintf("transferJobs/%s", parts[1]))
+	default:
+		return nil, fmt.Errorf("Invalid transfer job specifier. Expecting {projectId}/{transferJobName}")
+	}
+	return []*schema.ResourceData{d}, nil
+}
+
+func objectConditions() *schema.Schema {
```

Comment:
nit: Do you mind moving this + other subresource schemas right below the main schema?